include/deal.II/numerics/data_postprocessor.h

```diff
@@ -591,8 +595,169 @@ class DataPostprocessorScalar : public DataPostprocessor<dim>
+ *   data_out.add_data_vector (solution, gradient_postprocessor);
+ *   data_out.build_patches ();
+ *
+ *   std::ofstream output ("solution.vtu");
+ *   data_out.write_vtu (output);
+ * @endcode
+ *
+ * This leads to the following output for the solution and the
+ * gradients (you may want to compare with the solution shown in the results
+ * section of step-6; the current data is generated on a coarser mesh for
+ * simplicity):
+ *
+ * @image html data_postprocessor_vector_0.png
+ * @image html data_postprocessor_vector_1.png
+ *
+ * In the second image, the background color corresponds to the magnitude of the
+ * gradient vector and the vector glyphs to the gradient itself. It may be surprising
+ * at first to see that from each vertex, multiple vectors originate, going in
+ * different directions. But that is because the solution is only continuous:
+ * in general, the gradient is discontinuous across edges, and so the multiple
+ * vectors originating from each vertex simply represent the differing
+ * gradients of the solution at each adjacent cell.
+ *
+ * The output above -- namely, the gradient $\nabla u$ of the solution --
+ * corresponds to the "strain" in linear elasticity if the solution were scalar.
+ * It is very small in the central part of the domain because in step-6 we are
+ * solving an equation that has a coefficient $a(\mathbf x)$ that is large in
+ * the central part and small on the outside. On the other hand, the "stress"
+ * corresponds to the quantity $a(\mathbf x) \nabla u(\mathbf x)$. For the
+ * solution of that equation, the stress should be continuous and about as large
+ * in the interior part of the domain as on the part closer to the boundary.
+ * This is easily verified by the following modification of the postprocessor:
+ * @code
+ * template <int dim>
+ * class StressPostprocessor : public DataPostprocessorVector<dim>
+ * {
+ * public:
+ *   StressPostprocessor ()
+ *     :
+ *     // like above, but now also make sure that DataOut provides
+ *     // us with coordinates of the evaluation points:
+ *     DataPostprocessorVector<dim> ("stress",
+ *                                   update_gradients | update_quadrature_points)
+ *   {}
+ *
+ *   virtual
+ *   void
+ *   evaluate_scalar_field (const DataPostprocessorInputs::Scalar<dim> &input_data,
+ *                          std::vector<Vector<double> >               &computed_quantities) const
+ *   {
+ *     AssertDimension (input_data.solution_gradients.size(),
+ *                      computed_quantities.size());
+ *
+ *     for (unsigned int p=0; p<input_data.solution_gradients.size(); ++p)
+ *       {
+ *         AssertDimension (computed_quantities[p].size(), dim);
+ *         for (unsigned int d=0; d<dim; ++d)
+ *           // like above, but also multiply the gradients with
+ *           // the coefficient evaluated at the current point:
+ *           computed_quantities[p][d]
+ *             = input_data.solution_gradients[p][d];
```

Comment:
This doesn't seem to match the comment above. Are you missing the coefficient here?

Reply:
Yes, good spot!